packages/manager/modules/models/dedicated-server/hardware/hardware.class.js

```diff
@@ -0,0 +1,27 @@
+import { DedicatedServerHardwareSpecifications } from './specifications/specifications.class';
+
+export class DedicatedServerHardware {
+  constructor({ specifications } = {}) {
+    this.specifications = null;
+
+    if (specifications) {
+      this.setSpecifications(specifications);
+    }
+  }
+
+  setSpecifications(specifications = {}) {
+    if (specifications instanceof DedicatedServerHardwareSpecifications) {
+      this.specifications = specifications;
+    } else {
+      this.specifications = new DedicatedServerHardwareSpecifications(
+        specifications,
+      );
+    }
```

Comment:
I think we should expect specifications to be a DedicatedServerHardwareSpecifications and the DedicatedServerHardware shouldn't have to worry about creating the instance. 
If instance can't be created due to corrupted or missing property it shouldn't be the concern of this class 
Also are the specifications meant to change ? If not a setter is probably not necessary